docs/auth.md

```diff
@@ -0,0 +1,462 @@
+```
+
+### How to refresh a token?
+
+The access token will be valid only for 24 hours. After that, a new access
+token must be asked. To do that, just follow the refresh token flow, as
+explained [below](#post-authaccess_token).
+
+
+Devices and browser extensions
+------------------------------
+
+For devices and browser extensions, it is nearly the same than for third-party
+websites. The main difficulty is the redirect_uri. In OAuth2, the access code
+is given to the client by redirecting the user to an URL controlled by the
+client. But devices and browser extensions don't have an obvious URL for that.
+
+The IETF has published an RFC called [OAuth 2.0 for Native
+Apps](https://tools.ietf.org/html/draft-ietf-oauth-native-apps-05).
+
+### Native apps on desktop
+
+A desktop native application can start an embedded webserver on localhost. The
+redirect_uri will be something like `http://127.0.0.1:19856/callback`.
+
+### Native apps on mobile
+
+On mobile, the native apps can often register a custom URI scheme, like
+`com.example.oauthclient:/`. Just be sure that no other app has registered
+itself with the same URI.
+
+### Chrome extensions
+
+Chrome extensions can use URL like
+`https://<extension-id>.chromiumapp.org/<anything-here>` for their usage.
+See https://developer.chrome.com/apps/app_identity#non for more details. It
+has also a method to simplify the creation of such an URL:
+[`chrome.identity.getRedirectURL`](https://developer.chrome.com/apps/identity#method-getRedirectURL).
+
+### Firefox extensions
+
+It is possible to use an _out of band_ URN: `urn:ietf:wg:oauth:2.0:oob:auto`.
+The token is then extracted from the title of the page.
+See [this addon for google
+oauth2](https://github.com/AdrianArroyoCalle/firefox-addons/blob/master/addon-google-oauth2/addon-google-oauth2.js)
+as an example.
+
+
+Security considerations
+-----------------------
+
+The password will be stored in a secure fashion, with a password hashing
+function. The hashing function and its parameter will be stored with the hash,
+in order to make it possible to change the algorithm and/or the parameters
+later if we had any suspicion that it became too weak. The initial algorithm
+is [scrypt](https://godoc.org/golang.org/x/crypto/scrypt).
+
+The access code is valid only once, and will expire after 5 minutes
+
+Dynamically registered applications won't have access to all possible scopes.
+For example, an application that has been dynamically registered can't ask the
+cozy owner to give it the right to install other applications. This limitation
+should improve security, as avoiding too powerful scopes to be used with
+unknown applications.
+
+The cozy stack will apply rate limiting to avoid brute-force attacks.
+
+The cozy stack offers
+[CORS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS)
+for most of its services. But it's disabled for `/auth` (it doesn't make sense
```

Comment:
Also important to not CORS the other applications `index.html` as they contains tokens.